src/builder/FFI.sol

```diff
@@ -0,0 +1,24 @@
+// SPDX-License-Identifier: BSD-3-Clause
+pragma solidity ^0.8.27;
+
+import {IAcrossFFI} from "src/interfaces/IAcrossFFI.sol";
+
+/**
+ * @title Foreign Function Interface (FFI) Helper
+ * @notice Defines the addresses of reserved FFIs and methods for calling them
+ */
+library FFI {
+    /* FFI Addresses (starts from 0xFF1000, FFI with 100 reserved addresses) */
+    address constant ACROSS_FFI_ADDRESS = address(0xFF1000);
+
+    function requestAcrossQuote(
+        address inputToken,
+        address outputToken,
+        uint256 srcChain,
+        uint256 dstChain,
+        uint256 amount
+    ) internal pure returns (uint256 gasFee, uint256 variableFeePct) {
```

Comment:
curious if this is better to be a `view`. In better languages, improperly making something `pure` might allow optimizations to memoize the value, but given we know Solidity doesn't do that, I'm not too concern if we keep this `pure`, even knowing it has side-effects.

Reply:
Yeah, it shouldn't matter in Solidity, so no strong preference here